maint/ManyConfigTests

```diff
@@ -110,11 +110,12 @@ tmp=/tmp/pcre2testing
 
 CFLAGS="-g"
 OFLAGS="-O0"
+CC="${CC:=cc}"
 ISGCC=0
 
 # If the compiler is gcc, add a lot of warning switches.
 
-cc --version >/tmp/pcre2ccversion 2>/dev/null
+$CC --version >/tmp/pcre2ccversion 2>/dev/null
 if [ $? -eq 0 ] && grep GCC /tmp/pcre2ccversion >/dev/null; then
```

Comment:
Very curiously, on an Ubuntu machine, `cc --version` does not match "GCC", nor does `gcc --version` (it says "gcc" lowercase), nor does `clang --version`.

Hmm. I should check it's all OK in the CI!

Reply:
I found that the CI is not using all the GCC-specific warnings in the ManyConfigTests job! Oops. As you'd expect, the are some warnings that have crept in. I've made a separate PR for it: #670